RecoTracker/MeasurementDet/src/TkMeasurementDetSet.h

```diff
@@ -208,7 +208,7 @@ class StMeasurementDetSet {
   const edm::Handle<edmNew::DetSetVector<SiStripCluster>>& handle() const { return handle_; }
   // StripDetset & detSet(int i) { return detSet_[i]; }
   const StripDetset& detSet(int i) const {
-    if (detSet_[i].ready_)
+    if (detSet_[i].ready_ == ReadyState::kUnset)
```

Comment:
I believe this is wrong. This should be
```suggestion
    if (detSet_[i].ready_ != ReadyState::kSet)
```
as we need this to wait if it is in the middle of being set.

Reply:
Also, since we do not need sequential consistency here, maybe change to
```suggestion
    if (detSet_[i].ready_.load(std::memory_order_relaxed) !=ReadyState::kSet)
```

Reply:
> I believe this is wrong. This should be
> 
> as we need this to wait if it is in the middle of being set.

Fixed now.